### PR TITLE
build_providers: add ssh key managemet to the qemu build provider

### PR DESCRIPTION
**Pull request by sergiusens**
…ider

LP: #1774013
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [ ] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: snapcraft/internal/build_providers/_qemu/_keys.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+from paramiko import RSAKey
+
+from snapcraft.internal.errors import SnapcraftError
+
+
+SSHKeyT = TypeVar('SSHKeyT', bound='SSHKey')
+
+
+class SSHKeyPathError(SnapcraftError):
```
Comment (kyrofa) [inline snapcraft/internal/build_providers/_qemu/_keys.py]: This error class name could be more descriptive: it's not just a problem with the path (permissions etc.), it's that the path doesn't actually exist.

Comment (kyrofa) [reply snapcraft/internal/build_providers/_qemu/_keys.py]: Also, no strong feelings here, but I'm curious why this isn't in `build_providers/errors.py`?

File: snapcraft/internal/build_providers/_qemu/_keys.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        '{private_key_file_path!r} does not exist. '
+        'A private key is required.\n'
+        'Please file a report on https://launchpad.net/snapcraft/+filebug'
+    )
+
+    def __init__(self, *, private_key_file_path: str) -> None:
+        super().__init__(private_key_file_path=private_key_file_path)
+
+
+class SSHKey:
```
Comment (kyrofa) [inline snapcraft/internal/build_providers/_qemu/_keys.py]: Class and function docs would be nice.

Comment (sergiusens) [reply snapcraft/internal/build_providers/_qemu/_keys.py]: hmm, I wonder where they went, I did actually write them :confused:

Comment (sergiusens) [reply snapcraft/internal/build_providers/_qemu/_keys.py]: reflog did not rescue me :-/

File: snapcraft/internal/build_providers/_qemu/_keys.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/internal/build_providers/_qemu/_keys.py]: I had the feeling it was qemu specific, but after a check I see the telnet stuff made it there too.

**Pull request merged**